core/src/validator.rs

```diff
@@ -175,7 +179,37 @@ impl Validator {
             config.storage_slots_per_turn,
             bank.slots_per_segment(),
         );
-
+        if config.enable_ctrl_c_handler {
+            let bank_forks = bank_forks.clone();
+            let counter = Arc::new(AtomicUsize::new(0));
+            set_handler(move || {
+                if counter.fetch_add(1, Ordering::Relaxed) > 0 {
+                    warn!("Validator aborting on second ^C");
+                    std::process::exit(1);
+                }
+                let bank = bank_forks.read().unwrap().working_bank();
+                let _: Vec<_> = bank
+                    .epoch_vote_accounts(bank.get_epoch_and_slot_index(bank.slot()).0)
+                    .map(|accounts| {
+                        accounts
+                            .iter()
+                            .filter_map(|(pubkey, (stake, account))| {
+                                if let Ok(vote_state) = VoteState::deserialize(&account.data) {
```

Comment:
We only want to exit if _this_ validator has stake.  Right now this code will exit if any validator has stake.   you can compare `pubkey` with `keypair.pubkey()` and skip over deserializing the vote account of all validators except _this_ one.